packages/graph/hash_graph/lib/graph/src/store/postgres/ontology/data_type/mod.rs

```diff
@@ -79,36 +82,55 @@ impl<C: AsClient> DataTypeStore for PostgresStore<C> {
-        &self,
-        query: &DataTypeQuery,
-    ) -> Result<Vec<DataTypeRootedSubgraph>, QueryError> {
+    async fn get_data_type(&self, query: &DataTypeQuery) -> Result<Subgraph, QueryError> {
         let DataTypeQuery {
             ref expression,
-            data_type_query_depth,
+            query_resolve_depths,
         } = *query;
 
-        stream::iter(Read::<PersistedDataType>::read(self, expression).await?)
-            .then(|data_type| async move {
-                let mut referenced_data_types = DependencyMap::new();
-
-                self.get_data_type_as_dependency(
-                    data_type.identifier.uri(),
-                    DataTypeDependencyContext {
-                        referenced_data_types: &mut referenced_data_types,
-                        data_type_query_depth,
-                    },
-                )
+        let roots_and_vertices =
+            stream::iter(Read::<PersistedDataType>::read(self, expression).await?)
+                .then(|data_type| async move {
+                    let mut referenced_data_types = DependencyMap::new();
+
+                    self.get_data_type_as_dependency(
+                        data_type.identifier.uri(),
+                        DataTypeDependencyContext {
+                            referenced_data_types: &mut referenced_data_types,
+                            data_type_query_depth: query_resolve_depths.data_type_resolve_depth,
+                        },
+                    )
+                    .await?;
+
+                    let data_type = referenced_data_types
+                        .remove(data_type.identifier.uri())
+                        .expect("root was not added to the subgraph");
+
+                    return Ok::<_, Report<QueryError>>((
+                        GraphElementIdentifier::OntologyElementId(
+                            data_type.identifier.uri().clone(),
+                        ),
+                        (
+                            GraphElementIdentifier::OntologyElementId(
+                                data_type.identifier.uri().clone(),
+                            ),
+                            Vertex::DataType(data_type),
+                        ),
+                    ));
+                })
+                .try_collect::<Vec<_>>()
                 .await?;
 
-                let root = referenced_data_types
-                    .remove(data_type.identifier.uri())
-                    .expect("root was not added to the subgraph");
+        let (roots, vertices) = roots_and_vertices.into_iter().unzip();
 
-                Ok(DataTypeRootedSubgraph { data_type: root })
-            })
-            .try_collect()
-            .await
+        Ok(Subgraph {
+            roots,
+            vertices,
+            // TODO - we need to update the `DependencyMap` mechanism to collect these
+            //  https://app.asana.com/0/1203007126736604/1203160580911226/f
+            edges: HashMap::with_capacity(0),
```

Comment:
```suggestion
            edges: HashMap::new(),
```

Reply:
Oh wow I didn't even check the documentation (oops). Makes sense that Rust wouldn't allocate

> The hash map is initially created with a capacity of 0, so it will not allocate until it is first inserted into.

That's very different from other languages..